app/src/cose.h

```diff
@@ -121,92 +220,463 @@ namespace scitt::cose
+
+    QCBORDecode_EnterArray(&ctx, nullptr);
+    qcbor_result = QCBORDecode_GetError(&ctx);
+    if (qcbor_result != QCBOR_SUCCESS)
+    {
+      throw COSEDecodeError("Failed to parse COSE_Sign1 outer array");
+    }
+
+    uint64_t tag = QCBORDecode_GetNthTagOfLast(&ctx, 0);
+    if (tag != CBOR_TAG_COSE_SIGN1)
+    {
+      throw COSEDecodeError("COSE_Sign1 is not tagged");
+    }
+
+    auto phdr = decode_protected_header(ctx);
+    auto uhdr = decode_unprotected_header(ctx);
+
+    return std::make_tuple(phdr, uhdr);
+  }
+
   struct COSESignatureValidationError : public std::runtime_error
   {
     COSESignatureValidationError(const std::string& msg) :
       std::runtime_error(msg)
     {}
   };
 
+  // Temporarily needed for notary_verify().
+  bool is_ecdsa_alg(int64_t cose_alg)
+  {
+    return cose_alg == T_COSE_ALGORITHM_ES256 ||
+      cose_alg == T_COSE_ALGORITHM_ES384 || cose_alg == T_COSE_ALGORITHM_ES512;
+  }
+
+  // Temporarily needed for notary_verify().
+  bool is_rsa_pss_alg(int64_t cose_alg)
+  {
+    return cose_alg == T_COSE_ALGORITHM_PS256 ||
+      cose_alg == T_COSE_ALGORITHM_PS384 || cose_alg == T_COSE_ALGORITHM_PS512;
+  }
+
+  // Temporarily needed for notary_verify().
+  crypto::MDType get_md_type(int64_t cose_alg)
+  {
+    switch (cose_alg)
+    {
+      case T_COSE_ALGORITHM_ES256:
+      case T_COSE_ALGORITHM_PS256:
+        return crypto::MDType::SHA256;
+      case T_COSE_ALGORITHM_ES384:
+      case T_COSE_ALGORITHM_PS384:
+        return crypto::MDType::SHA384;
+      case T_COSE_ALGORITHM_ES512:
+      case T_COSE_ALGORITHM_PS512:
+        return crypto::MDType::SHA512;
+      case T_COSE_ALGORITHM_EDDSA:
+        return crypto::MDType::NONE;
+      default:
+        throw std::runtime_error("Unsupported COSE algorithm");
+    }
+  }
+
+  // Temporarily needed for notary_verify().
+  const EVP_MD* get_openssl_md_type(crypto::MDType type)
+  {
+    switch (type)
+    {
+      case crypto::MDType::NONE:
+        return nullptr;
+      case crypto::MDType::SHA1:
+        return EVP_sha1();
+      case crypto::MDType::SHA256:
+        return EVP_sha256();
+      case crypto::MDType::SHA384:
+        return EVP_sha384();
+      case crypto::MDType::SHA512:
+        return EVP_sha512();
+      default:
+        throw std::runtime_error("Unsupported hash algorithm");
+    }
+    return nullptr;
+  }
+
+  // Temporarily needed for notary_verify().
+  std::vector<uint8_t> create_sign1_tbs(
+    int cose_size,
+    std::span<const uint8_t> protected_header,
+    std::span<const uint8_t> payload)
+  {
+    // Note: This function does not return the hash of the TBS because
+    // EdDSA does not support pre-hashed messages as input.
+
+    // Sig_structure = [
+    //     context: "Signature1",
+    //     body_protected: empty_or_serialized_map,
+    //     external_aad: bstr,
+    //     payload: bstr
+    // ]
+
+    cbor::encoder ctx(cose_size + 1024);
```

Comment:
`protected_header.size() + payload.size() + 1024` seems good enough (if anything it's more accurate), no need to have a `cose_size` parameter.

Reply:
Done in https://github.com/microsoft/scitt-ccf-ledger/pull/90